tests/func/test_api.py

```diff
@@ -47,10 +46,10 @@ def test_get_url_external(erepo_dir, cloud):
 def test_get_url_requires_dvc(tmp_dir, scm):
     tmp_dir.scm_gen({"foo": "foo"}, commit="initial")
 
-    with pytest.raises(UrlNotDvcRepoError, match="not a DVC repository"):
+    with pytest.raises(OutputNotFoundError, match="with output 'foo'"):
```

Comment:
This is changing the API, we shouldn't do it in this PR if we can avoid it.

Reply:
Plus `OutputNotFoundError` is an internal thing, that shouldn't be exposed to api users.

Reply:
We do have to break things here as it's not precise enough. OutputNotFoundError is more of a correct term here.

Reply:
Also, we already threw `OutputNotFoundError`. It was just not documented. The following threw `OutputNotFoundError`.

https://github.com/iterative/dvc/blob/334556f07dc511927543218d4a2a1a1c1c83ed65/dvc/api.py#L31

Reply:
@skshetry I think that one was caught by https://github.com/iterative/dvc/blob/master/dvc/external_repo.py#L51 and re-raised properly.

Reply:
Ahh missed that. So, we could raise the same exc here?

Reply:
@skshetry But this particular test doesn't catch that one, but rather UrlNotDvcRepoError and it should stay that way. The issue is that we are using RepoTree instead of DvcTree right now, which works for git-only repos, when it shouldn't. Could do something simple like `if not tree.dvc_tree: raise UrlNotDvcRepoError` though, if subrepos are a concern.